### PR TITLE
specifying which data to return from optimisation

### PR DESCRIPTION
**Pull request by HaraldGSvendsen**
Implemented possibility to specify which data to return (defaults to all)

### CODE REVIEW

File: src/oogeso/core/simulator.py [hunk 1]
```diff
@@ -54,30 +56,56 @@ def __init__(self, data: EnergySystemData):
...
-        store_duals=None,
-        write_yaml=False,
-    ):
-        """Solve problem over many timesteps - rolling horizon
+        solver: str,
+        timerange: Tuple[int, int] = None,
+        timelimit: int = None,
+        return_variables: Sequence[str] = None,
+        store_duals: dict = None,
+        write_yaml: bool = False,
```
Comment (mortendaehli) [inline src/oogeso/core/simulator.py:64]: Are these optional since they default to None?

File: src/oogeso/core/simulator.py [hunk 2]
```diff
@@ -112,36 +140,65 @@ def runSimulation(
...
         # hdf5? https://stackoverflow.com/questions/47072859/how-to-append-data-to-one-specific-dataset-in-a-hdf5-file-with-h5py)
         pyomo_instance = self.optimiser.pyomo_instance
         timelimit = self.optimiser.optimisation_parameters.optimisation_timesteps
-        data_to_keep = self.optimiser.optimisation_parameters.optimisation_return_data
-        timeshift = timestep
 
-        # TODO: Implement possibility to store subset of the data
-        if data_to_keep is not None:
-            logger.warning("Storing only a subset of the data not implemented yet.")
+        all_available_variables = [
```
Comment (mortendaehli) [inline src/oogeso/core/simulator.py:161]: A more generic and simpler way to do this is to instead do for each if-statement do

Set return_variables as Optional

return_all = False
if not return_variables:
   return_all = True

<code here>

if return_all or "dfDeviceFlow" in return_variables:
    <code here>
if return_all or dfDeviceIsPrep in retur_variables:
    <code here>
    
and so on...

I do not see a need to change this now, but next time you make a change to this then that would make it easier in the future to maintain and update.

File: src/oogeso/dto/oogeso_output_data_objects.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ class SimulationResult:
     # Max available "flow" (power/fluid) from storage (Sm3/s or MW):
     dfDeviceStoragePmax: pd.Series = None
     # Device assosiated penalty rate (PENALTY_UNIT/s):
-    dfPenalty: pd.DataFrame = None
+    dfPenalty: pd.Series = None
```
Comment (mortendaehli) [inline src/oogeso/dto/oogeso_output_data_objects.py:35]: It is sufficient to do dfPenalty: pd.Series unless you want it optional. Then you should do df: Optional[pd.Series] = None

File: src/oogeso/plots/plots.py [hunk 4]
```diff
@@ -99,7 +99,7 @@ def plot_deviceprofile(
         nrows = nrows + 1
     df2 = res.dfDeviceIsOn.unstack("device")[devs]
     dfPrep = res.dfDeviceIsPrep.unstack("device")[devs]
-    timerange = list(res.dfExportRevenue.index)
+    timerange = list(res.dfDeviceIsOn.index.get_level_values("time"))
```
Comment (mortendaehli) [inline src/oogeso/plots/plots.py:102]: Just a note for the future. Python convention is snake case for instances of a class: df_device_is_on, etc. Makes it easier to read :)

Comment (mortendaehli) [reply src/oogeso/plots/plots.py:102]: No need to change anything for now. Just an advice for the future.

**Review by mortendaehli — APPROVED** [review]

**Comment by mortendaehli** [comment]
Looks good. Just a few tips on code convention and style.

File: src/oogeso/core/simulator.py (hunk 1 above)
Comment (HaraldGSvendsen) [reply src/oogeso/core/simulator.py:64]: Yes, optional. Have added Optional[...]

File: src/oogeso/plots/plots.py (hunk 4 above)
Comment (HaraldGSvendsen) [reply src/oogeso/plots/plots.py:102]: OK, noted

File: src/oogeso/dto/oogeso_output_data_objects.py (hunk 3 above)
Comment (HaraldGSvendsen) [reply src/oogeso/dto/oogeso_output_data_objects.py:35]: OK, removed default None value

**Pull request merged**